packages/mgt-chat/src/statefulClient/StatefulGraphChatClient.ts

```diff
@@ -493,7 +493,10 @@ class StatefulGraphChatClient extends BaseStatefulClient<GraphChatClient> {
       // load delta messages
       const deltaMessages = await this.loadDeltaData(this._chatId, cachedMessages.lastModifiedDateTime);
       // add delta messages to cache
-      const updatedState = await this._cache.cacheMessages(this._chatId, deltaMessages);
+      // Filter out duplicate messages
```

Comment:
@Mnickii A little curious as to what causes messages to be duplicated?

Reply:
Sure, there are issues with duplicate/wrong caching of chat messages, which seem to result from graph call times. This fix improves the chat message accuracy but there's now more duplication.

Reply:
By more duplication you mean a message is now displayed a number of times while initially it was like once?

Reply:
Yes exactly